lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,781 @@
+        break;
+        case BTOR2_TAG_smod: 
+            res = builder.create<btor::SModOp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_xnor: 
+            res = builder.create<btor::XnorOp>(unknownLoc, 
+                                            cache.at(kids[0]), 
+                                            cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_xor: 
+            res = builder.create<btor::XOrOp>(unknownLoc, 
+                                            cache.at(kids[0]), 
+                                            cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_sll: 
+            res = builder.create<btor::ShiftLLOp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_sra: 
+            res = builder.create<btor::ShiftRAOp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_srl: 
+            res = builder.create<btor::ShiftRLOp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_rol: 
+            res = builder.create<btor::RotateLOp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_ror: 
+            res = builder.create<btor::RotateROp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_saddo: 
+            res = builder.create<btor::SAddOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_sdivo: 
+            res = builder.create<btor::SDivOverflowOp>(unknownLoc,
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_smulo: 
+            res = builder.create<btor::SMulOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_ssubo: 
+            res = builder.create<btor::SSubOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
```

Comment:
wondering why is integer type needed here as parameter. seems unnecessary...

Reply:
The return type of this operation is `i1`, however, the input is not restricted to this requirement